src/processing/action.py

```diff
@@ -1,45 +1,103 @@
 import torch
-from state import GameState, Frame, PlayerFrame, Keypoint
-from pose_estimation.pose_estimate import KeyPointNames, AngleNames
-
-COMBINATIONS = AngleNames.combinations
-ANGLE_NAMES = AngleNames.list
+from state import GameState, PlayerFrame, BallFrame, Box
+from args import DARGS
 
 class ActionRecognition:
-    def __init__(self, state: GameState):
+    def __init__(self, state: GameState, args=DARGS) -> None:
         self.state = state
+        self.THRESHOLD = args["shot_threshold"]
+        self.ANGLE_THRESHOLD = args["angle_threshold"]
 
-    @staticmethod
-    def is_shot(player_frame: PlayerFrame):
+    def pose_shot(self, player_frame: PlayerFrame, ANGLE_THRESHOLD):
         """
-        Takes in keypoint and angle data for a player in a frame and returns whether
-        or not the player is shooting. Currently uses a simple threshold heuristic.
+        Takes in keypoint and angle data for a player in a frame and returns the
+        weighted sum of the pose features. Currently uses a simple threshold
+        heuristic.
         """
+        keypoints = ["left_wrist", "right_wrist", "left_shoulder", "right_shoulder"]
+        for keypoint in keypoints:
+            if not keypoint in player_frame.keypoints:
+                return False
         # Assuming keypoints are in the form {name: Keypoint}
-        left_wrist = torch.tensor([player_frame.keypoints['left_wrist'].x, player_frame.keypoints['left_wrist'].y])
-        right_wrist = torch.tensor([player_frame.keypoints['right_wrist'].x, player_frame.keypoints['right_wrist'].y])
-        left_shoulder = torch.tensor([player_frame.keypoints['left_shoulder'].x, player_frame.keypoints['left_shoulder'].y])
-        right_shoulder = torch.tensor([player_frame.keypoints['right_shoulder'].x, player_frame.keypoints['right_shoulder'].y])
-
-        left_knee = player_frame.angles('left_knee')
-        right_knee = player_frame.angles('right_knee')
-        left_elbow = player_frame.angles('left_elbow')
-        right_elbow = player_frame.angles('right_elbow')
-
-        THRESHOLD = 0.7 # TODO abstract into config file
-        ANGLE_THRESHOLD = 150
+        left_wrist = torch.tensor(
```

Comment:
write a method inside Keypoints class to do this for you? Like a to_tensor() method.